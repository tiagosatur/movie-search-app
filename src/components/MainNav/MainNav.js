import React from 'react'
import { Link } from 'react-router-dom';
import styled, { css } from 'styled-components';

import { PATHS } from '../../utils/variables';
import { colorPallete, rem } from '../../style'


const MainNav = () => (
    <StyledNav>
        <StyledLogo>
            <Link to={PATHS.HOME.PATH}><h1>MovieFlix</h1></Link>
        </StyledLogo>
        <StyledNavList>
            {
                Object.keys(PATHS).map((item) => (
                    
                    <li key={PATHS[item].PATH}>
                        <Link to={PATHS[item].PATH}>{PATHS[item].LABEL}</Link>
                    </li>
                ))
            }
        </StyledNavList>
    </StyledNav>
);

const StyledNav = styled.nav`
    background: ${ colorPallete.primaryGradient };
    display: flex;
    justify-content: center;
    flex-direction: column;
    padding: ${rem(32)};

    @media(min-width: 768px) {
        justify-content: space-between;
        flex-direction: row;
    }
`;

const StyledNavList = styled.ul`
    align-items: center;
    display: flex;
    justify-content: flex-end;
    list-style: none;
    margin: 0;
    padding: 0;

    li {
        a {
            color: ${ colorPallete.white };
            font-size: ${rem(16)};
            font-weight: 200;
            outline: none;
            padding: ${rem(8)} ${rem(10)};
            position: relative;

            &:after {
                background: white;
                bottom: ${rem(-5)};
                content: '';
                position: absolute;
                height: ${rem(2)};
                left: 0;
                transition: width .4s;
                width: 0;

            }

            &:hover, &:focus {
                
                &:after {
                    width: 95%;
                }
            }
        }
    }
`;

const StyledLogo = styled.div`
    font-family: 'Kanit';
    font-size: ${rem(24)};
    font-weight: bold;

    h1 {
        margin: 0;
    }

    a {
        color: ${ colorPallete.white };
        text-decoration: none;
    }
`;

export default MainNav;