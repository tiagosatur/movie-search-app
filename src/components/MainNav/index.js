import React from 'react'
import { Link } from 'react-router-dom';
import { PATHS } from '../../utils/variables';
const ulStyle = {
    display: 'flex',
    listStyle: 'none',
};
const liStyle = {padding: '4px'};

const MainNav = () => (
    <nav>
        <ul style={ulStyle}>
            {
                Object.keys(PATHS).map((item) => (
                    
                    <li key={PATHS[item].PATH} style={liStyle}>
                        <Link to={PATHS[item].PATH}>{PATHS[item].LABEL}</Link>
                    </li>
                ))
            }
        </ul>
    </nav>
);

export default MainNav;