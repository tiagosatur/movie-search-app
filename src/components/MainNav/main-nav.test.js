import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MainNav from './MainNav';

describe('MainNav tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MainNav />
    );
  });
  
  it('should render the MainNav element', () => {
    expect(wrapper.find('.main-nav').exists()).toBe(true);
  });

  it('should call MainNav snapshot correctly', () => {
    const tree = renderer
        .create(wrapper)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });
})