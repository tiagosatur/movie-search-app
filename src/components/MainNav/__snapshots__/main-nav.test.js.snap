// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav tests should call MainNav snapshot correctly 1`] = `null`;
